cli/sawtooth_cli/block.py

```diff
@@ -46,13 +46,20 @@ def add_block_parser(subparsers, parent_parser):
         'blockchain, including the block id and number, public keys all '
         'of allsigners, and number of transactions and batches.')
 
-    grand_parsers.add_parser(
+    list_parser = grand_parsers.add_parser(
         'list',
         help='Displays information for all blocks on the current blockchain',
         description=description,
         parents=[base_http_parser(), base_list_parser()],
         formatter_class=argparse.RawDescriptionHelpFormatter)
 
+    list_parser.add_argument(
+        '--limit',
+        type=str,
+        help='the number of blocks to list',
+        default="100"
```

Comment:
How do I handle the unlimited case?  I.e. what if I want to fetch all and just stop when I find what I'm looking for?

Reply:
I will remove the default limit of 100 and it if limit is not provided, it will return like it does currently.